podpac/core/algorithm/stats.py

```diff
@@ -916,14 +892,25 @@ def eval(self, coordinates, output=None):
             # standard, e.g. grouped.median('time')
             out = getattr(grouped, self.reduce_fn)("time")
 
-        # map
-        eval_time = xr.DataArray(coordinates.coords["time"])
-        E = getattr(eval_time.dt, self.groupby)
-        out = out.sel(**{self.groupby: E}).rename({self.groupby: "time"})
-        output[:] = out.transpose(*output.dims).data
+        out = out.rename({self.groupby: "time"})
+        if output is None:
+            coords = podpac.coordinates.merge_dims(
+                [coordinates.drop("time"), Coordinates([out.coords["time"]], ["time"])]
+            )
+            coords = coords.transpose(*out.dims)
+            output = self.create_output_array(coords, data=out.data)
+        else:
+            output.data[:] = out.data[:]
+
+        ## map
```

Comment:
@jmilloy : I completely changed how this node work... I figured that was okay because I suddenly had an actual use-case and there were no unit tests. :)

Reply:
How do you feel about still leaving it without tests?

Reply:
Not great. I'm planning to add some tests, before finishing this PR.

Reply:
If you have bandwidth, that would be great if you could write some tests. I'm also trying to finish up the parallel manager branch's tests.

Reply:
Yeah, I'll check in with you before I starting writing tests so we don't duplicate effort, but I might be able to get to it before you.